src/Proto.Mailbox/Mailbox.cs

```diff
@@ -115,10 +115,9 @@ private Task RunAsync()
                     _stats[i].MailboxEmpty();
                 }
             }
-            return Task.FromResult(0);
         }
 
-        private bool ProcessMessages()
+        private async Task<bool> ProcessMessages()
```

Comment:
It needs to be checked twice. Once for the case when the task runs synchronously. If it ran asynchronously then it needs to be checked in the continuation instead.